integrations/gcp/gcp_core/search/resource_searches.py

```diff
@@ -38,144 +45,157 @@ async def search_all_resources(
 async def search_all_resources_in_project(
     project: dict[str, Any], asset_type: str, asset_name: str | None = None
 ) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    """
-    List of supported assets: https://cloud.google.com/asset-inventory/docs/supported-asset-types
-    Search for resources that the caller has ``cloudasset.assets.searchAllResources`` permission on within the project's scope.
-    """
-    project_name = project["name"]
-    logger.info(f"Searching all {asset_type}'s in project {project_name}")
-    async with AssetServiceAsyncClient() as async_assets_client:
+
+    def parse_asset_response(response: Any) -> list[dict[Any, Any]]:
+        assets = typing.cast(list[AssetData], parse_protobuf_messages(response.results))
+        latest_resources = [
+            {
+                **parse_latest_resource_from_asset(asset),
+                EXTRA_PROJECT_FIELD: project,
+            }
+            for asset in assets
+        ]
+        return latest_resources
+
+    async with semaphore:
+        """
+        List of supported assets: https://cloud.google.com/asset-inventory/docs/supported-asset-types
+        Search for resources that the caller has ``cloudasset.assets.searchAllResources`` permission on within the project's scope.
+        """
+        project_name = project["name"]
+        logger.info(f"Searching all {asset_type}'s in project {project_name}")
+
         search_all_resources_request = {
             "scope": project_name,
             "asset_types": [asset_type],
             "read_mask": "*",
         }
         if asset_name:
             search_all_resources_request["query"] = f"name={asset_name}"
-        try:
-            paginated_responses: pagers.SearchAllResourcesAsyncPager = (
-                await async_assets_client.search_all_resources(
-                    search_all_resources_request, retry=retry_policy
+
+        asset_rate_limiter = await search_all_resources_qpm_per_project[
+            project_name.split("/")[-1]
+        ]
+
+        async with AssetServiceAsyncClient() as async_assets_client:
+
+            try:
+                async for assets in paginated_query(
+                    async_assets_client,
+                    "search_all_resources",
+                    search_all_resources_request,
+                    parse_asset_response,
+                    rate_limiter=asset_rate_limiter,
+                ):
+                    yield assets
+
+            except PermissionDenied:
+                logger.error(
+                    f"Service account doesn't have permissions to list topics from project {project_name}"
+                )
+            except NotFound:
+                logger.debug(
+                    f"Couldn't perform search_all_resources on project {project_name} since it's deleted."
+                )
```

Comment:
how this could be? that we are able to fetch it but not list it? I think worth having it as `info` as we do have this log that we start fetching from it

Reply:
Agreed that these logs should be info logs